src/js/containers/search/visualizations/rank/SpendingByCategoryRankVisualizationSectionContainer.jsx

```diff
@@ -0,0 +1,251 @@
+    objectClasses: 'object_class__object_class_name'
+};
+
+export class SpendingByCategoryRankVisualizationSectionContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+            labelSeries: [],
+            dataSeries: [],
+            descriptions: [],
+            linkSeries: [],
+            page: 1,
+            scope: 'budgetFunctions',
+            next: '',
+            previous: '',
+            hasNextPage: false,
+            hasPreviousPage: false
+        };
+
+        this.changeScope = this.changeScope.bind(this);
+        this.nextPage = this.nextPage.bind(this);
+        this.previousPage = this.previousPage.bind(this);
+        this.apiRequest = null;
+    }
+
+    componentDidMount() {
+        this.newSearch();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!_.isEqual(prevProps.reduxFilters, this.props.reduxFilters)
+            || (prevProps.budgetFiltersSelected !== this.props.budgetFiltersSelected)
+            || (prevProps.awardFiltersSelected !== this.props.awardFiltersSelected)) {
+            this.newSearch();
+        }
+    }
+
+    changeScope(scope) {
+        this.setState({
+            scope,
+            page: 1
+        }, () => {
+            this.fetchData();
+        });
+    }
+
+    newSearch() {
+        this.setState({
+            page: 1
+        }, () => {
+            this.fetchData();
+        });
+    }
+
+    nextPage() {
+        this.setState({
+            page: this.state.page + 1
+        }, () => {
+            this.fetchData();
+        });
+    }
+
+    previousPage() {
+        // change the state by subtracting 2 (since the page number is already incremented)
+        const prevPage = _.max([1, this.state.page - 1]);
+        this.setState({
+            page: prevPage
+        }, () => {
+            this.fetchData();
+        });
+    }
+
+    createTransactionsParams(operation, group) {
+        const searchParams = operation.toParams();
+
+        // generate the API parameters
+        const apiParams = {
+            field: 'federal_action_obligation',
+            group,
+            order: ['-aggregate'],
+            aggregate: 'sum',
+            filters: searchParams,
+            limit: 5,
+            page: this.state.page
+        };
+
+        return apiParams;
+    }
+
+    createTasParams(group) {
+        const operation = new SearchAccountOperation('tas');
+        operation.fromState(this.props.reduxFilters);
+
+        const searchParams = operation.toParams();
+
+        // generate the API parameters
+        const apiParams = {
+            field: 'obligations_incurred_by_program_object_class_cpe',
+            group,
+            order: ['-aggregate'],
+            aggregate: 'sum',
+            filters: searchParams,
+            limit: 5,
+            page: this.state.page
+        };
+
+        return apiParams;
+    }
+
+    fetchData() {
```

Comment:
Let's just make two separate fetch for each type of request rather than doing a lot of complex if/logic mixing (same applies to TimeVisualization....Container)